crates/block-producer/src/server.rs

```diff
@@ -25,95 +23,86 @@ use crate::{
     batch_builder::BatchBuilder,
     block_builder::BlockBuilder,
     domain::transaction::AuthenticatedTransaction,
-    errors::{AddTransactionError, BlockProducerError, VerifyTxError},
+    errors::{AddTransactionError, BlockProducerError, StoreError, VerifyTxError},
     mempool::{BatchBudget, BlockBudget, Mempool, SharedMempool},
     store::StoreClient,
 };
 
-/// Represents an initialized block-producer component where the RPC connection is open,
-/// but not yet actively responding to requests.
-///
-/// Separating the connection binding from the server spawning allows the caller to connect other
-/// components to the store without resorting to sleeps or other mechanisms to spawn dependent
-/// components.
+/// Block producer's configuration.
```

Comment:
This could explain how the struct works and how it expects to be configured for communication with other components (similar to how it was before)